### PR TITLE
Remove minimal implementations

### LINKED ISSUE #3276: Remove the MinimalImplementations.IAsyncDisposable class and use Microsoft.Bcl.AsyncInterfaces instead
Prior to the TFM change, we should remove the MinimalImplementations.IAsyncDisposable class and add a NuGet reference to Microsoft.Bcl.AsyncInterfaces instead.

_Originally posted by @sharwell in https://github.com/dotnet/roslyn-analyzers/pull/3273_

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #3276 

I have the following 2 errors that I don't manage to get over:

1. `error CS0433: The type 'HttpClient' exists in both 'System.Net.Http, Version=4.1.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' and 'System.Net.Http, Version=4.2.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'`

2. `error CS0234: The type or namespace name 'Security' does not exist in the namespace 'Microsoft' (are you missing an assembly reference?)`

### CODE REVIEW

**Comment by mavasani** [comment]
Also tagging @dotpaul for review

**Comment by Evangelink** [comment]
Aaaah! I keep forgetting about the security rules...

**Comment by dotpaul** [comment]
Other than targeting the v2.9.x branch, looks fine to me.

**Comment by Evangelink** [comment]
I'll update this PR once #3284 got merged.

**Comment by Evangelink** [comment]
@mavasani PR is ready and no longer blocked.

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Security/DoNotDisableHttpClientCRLCheck.cs [hunk 1]
```diff
@@ -105,8 +105,7 @@ class DoNotDisableHttpClientCRLCheck : DiagnosticAnalyzer
 
         private static readonly ImmutableHashSet<string> typeToTrackMetadataNames = ImmutableHashSet.Create<string>(
             WellKnownTypeNames.SystemNetHttpWinHttpHandler,
-            WellKnownTypeNames.SystemNetHttpHttpClientHandler,
-            WellKnownTypeNames.SystemNetHttpCurlHandler);
```
Comment (mavasani) [inline src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Security/DoNotDisableHttpClientCRLCheck.cs]: Can you please merge latest master into this branch?

Comment (Evangelink) [reply src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Security/DoNotDisableHttpClientCRLCheck.cs]: Done

**Comment by Evangelink** [comment]
ping @mavasani

File: src/Test.Utilities/AdditionalMetadataReferences.cs [hunk 2]
```diff
@@ -45,6 +46,18 @@ public static class AdditionalMetadataReferences
                 new PackageIdentity("Microsoft.AspNetCore.Mvc", "1.1.8"),
                 new PackageIdentity("Microsoft.AspNetCore.Http", "1.1.2")));
 
+        public static ReferenceAssemblies DefaultWithNUnit { get; } = Default
+            .AddPackages(ImmutableArray.Create(new PackageIdentity("NUnit", "3.12.0")));
+
+        public static ReferenceAssemblies DefaultWithxunit { get; } = Default
```
Comment (mavasani) [inline src/Test.Utilities/AdditionalMetadataReferences.cs]:
```suggestion
        public static ReferenceAssemblies DefaultWithXunit { get; } = Default
```

Comment (mavasani) [reply src/Test.Utilities/AdditionalMetadataReferences.cs]: Maybe use `DefaultWithXUnit` to keep the name consistent with `DefaultWithNUnit`?

Comment (Evangelink) [reply src/Test.Utilities/AdditionalMetadataReferences.cs]: Fine by me. FYI I went with this name to follow the library naming convention.

Comment (mavasani) [reply src/Test.Utilities/AdditionalMetadataReferences.cs]: Ah, then it is okay to leave it as-is.

**Pull request merged**